### PR TITLE
docs(splunk service): added highlight article for Splunk HEC improvements

### PR DESCRIPTION
**Pull request by barieom**
Highlight article for V0.19.0 release for:
- Splunk HEC token passthrough routing #10261 
- Splunk HEC indexer acks #10135

### CODE REVIEW

File: website/content/en/highlights/2021-12-15-splunk-hec-improvements.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+better end-to-end acknowledgement to ensure that no data is lost. As you may be
+aware, Splunk HEC does not guarantee that a data is successfully written and saved
+if it receives a 200. Vector will now check [Splunk HEC indexer acknowledgements][Splunk indexer]
+protocol to verify that data has been successfully delivered. To learn more about
+how it works, check out the short explanation in the Vector [sinks reference page][indexer how it works].
+
+Vector now also supports new token passthrough routing. You can now use these tokens,
+which can be found in `splunk_hec` source requests, downstream `splunk_hec` sink
+requests. With the ability to record Splunk token that the events came in with, you
+can now use that token in the `splunk_hec` sink to forward logs. With this update,
```
Comment (001wwang) [inline website/content/en/highlights/2021-12-15-splunk-hec-improvements.md:27]: This seems slightly repetitive with the previous sentence. Perhaps combine them into something like: 
> "When `store_hec_token` is enabled in a `splunk_hec` source, tokens included in requests to the source will be stored and used by downstream `splunk_hec` sinks. Any passed through token takes precedence over the `default_token` configuration set in the sink." 

Side note: the token passthrough works for both `splunk_hec` metrics and logs sinks so "to foward logs" may be a bit confusing.

File: website/content/en/highlights/2021-12-15-splunk-hec-improvements.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+you can now partition event batches in the `splunk_hec` sinks by token.
```
Comment (001wwang) [inline website/content/en/highlights/2021-12-15-splunk-hec-improvements.md:28]: I think you can leave this partitioning bit out as it's more of a minor implementation detail.

File: website/content/en/highlights/2021-12-15-splunk-hec-improvements.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+hide_on_release_notes: false
+badges:
+  type: announcement
+---
+
+We're excited to share that we've made a couple improvements to Vector's support
+for Splunk HEC; Vector now supports Splunk HEC indexer acknowledgements and
+token passthrough routing.
+
+With the new support for Splunk HEC indexer acknowledgement, Vector now provides a
```
Comment (binarylogic) [inline website/content/en/highlights/2021-12-15-splunk-hec-improvements.md:17]:
```suggestion
## Indexer Acknowledgements

With the new support for Splunk HEC indexer acknowledgement, Vector now provides a
```

File: website/content/en/highlights/2021-12-15-splunk-hec-improvements.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+better end-to-end acknowledgement to ensure that no data is lost. As you may be
+aware, Splunk HEC does not guarantee that a data is successfully written and saved
+if it receives a 200. Vector will now check [Splunk HEC indexer acknowledgements][Splunk indexer]
+protocol to verify that data has been successfully delivered. To learn more about
+how it works, check out the short explanation in the Vector [sinks reference page][indexer how it works].
+
+Vector now also supports new token passthrough routing. You can now use these tokens,
```
Comment (binarylogic) [inline website/content/en/highlights/2021-12-15-splunk-hec-improvements.md:24]:
```suggestion
## Passthrough Token Routing

Vector now also supports new token passthrough routing. You can now use these tokens,
```

**Review by 001wwang — COMMENTED** [review]
Awesome! Left some comments mostly on rewording and whether we'd want to highlight the `splunk_hec` source changes as well. Happy to chat about anything.

File: website/content/en/highlights/2021-12-15-splunk-hec-improvements.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+authors: ["barieom"]
+pr_numbers: [10261, 10135]
+release: "0.19.0"
+hide_on_release_notes: false
+badges:
+  type: announcement
+---
+
+We're excited to share that we've made a couple improvements to Vector's support
+for Splunk HEC; Vector now supports Splunk HEC indexer acknowledgements and
```
Comment (jszwedko) [inline website/content/en/highlights/2021-12-15-splunk-hec-improvements.md:14]: I might add some links to the Splunk HEC docs for people who are familiar with Splunk, but not the HEC format.

File: website/content/en/highlights/2021-12-15-splunk-hec-improvements.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+token passthrough routing.
+
+With the new support for Splunk HEC indexer acknowledgement, Vector now provides a
+better end-to-end acknowledgement to ensure that no data is lost. As you may be
+aware, Splunk HEC does not guarantee that a data is successfully written and saved
+if it receives a 200. Vector will now check [Splunk HEC indexer acknowledgements][Splunk indexer]
```
Comment (jszwedko) [inline website/content/en/highlights/2021-12-15-splunk-hec-improvements.md:20]:
```suggestion
if it responds to the incoming request successfully. Vector will now check [Splunk HEC indexer acknowledgements][Splunk indexer]
```

File: website/content/en/highlights/2021-12-15-splunk-hec-improvements.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+protocol to verify that data has been successfully delivered. To learn more about
+how it works, check out the short explanation in the Vector [sinks reference page][indexer how it works].
```
Comment (jszwedko) [inline website/content/en/highlights/2021-12-15-splunk-hec-improvements.md:22]:
```suggestion
how it works, check out the short explanation in the Vector [`splunk_hec` sink reference page][indexer how it works].
```

**Review by jszwedko — COMMENTED** [review]
Looks good!

I agree with Will that we should highlight the improvements to the `splunk_hec` source as well. These would be:

* The `splunk_hec` source now implements the Splunk HEC ack protocol to notify senders when the data has been successfully processed
* The `splunk_hec` source is now integrated into Vector's end-to-end acknowledgments so that it only acks events that have been processed by sinks (or disk buffers)

I might also add some configuration snippets to show how to enable and use these features.

**Comment by jszwedko** [comment]
@barieom I think this one still needs revised based on the feedback. I'm happy to take another look after that!

**Comment by jszwedko** [comment]
@001wwang could you give this another look over?

File: website/content/en/highlights/2021-12-15-splunk-hec-improvements.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+## Indexer Acknowledgements
+
+With the new support for Splunk HEC indexer acknowledgement, Vector is now able
+to provide higher guarantees that no data is lost when using the `splunk_hec`
+source and sink.
+
+As you may be aware, Splunk HEC does not guarantee that data is successfully
+written when responds to the incoming request successfully, but rather specifies
+that the acknowledgement status of sent events is provided via a [separate
+endpoint][indexer].
```
Comment (001wwang) [inline website/content/en/highlights/2021-12-15-splunk-hec-improvements.md:26]:
```suggestion
written when it responds to an incoming request successfully. To confirm that data has been persisted, the [indexer acknowledgements][indexer] feature must be enabled. The status of sent events can then be queried via a separate endpoint.
```

File: website/content/en/highlights/2021-12-15-splunk-hec-improvements.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+### `splunk_hec` source acknowledgements
+
+Previously, the `splunk_hec`  source did not support [indexer
+acknowledgements][indexer] and so would provide weaker delivery guarantees and
+would not work with Splunk senders that required them.
+
+Now, you can configure the `splunk_hec` source to responding to indexer
+acknowledgement requests from clients by configuring:
```
Comment (001wwang) [inline website/content/en/highlights/2021-12-15-splunk-hec-improvements.md:35]:
```suggestion
Now, you can configure the `splunk_hec` source to use the indexer acknowledgements protocol by configuring:
```

File: website/content/en/highlights/2021-12-15-splunk-hec-improvements.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+```toml
+acknowledgements = true
+```
+
+In addition to this configuring the source to expose `/services/collector/ack`
+for querying for ack status by clients, this also wires into Vector's
```
Comment (001wwang) [inline website/content/en/highlights/2021-12-15-splunk-hec-improvements.md:42]:
```suggestion
When enabled, responses to incoming requests will include an ID that can be used to query for acknowledgement status at the newly exposed `/services/collector/ack` endpoint ([learn more here][indexer how it works]). The acknowledgement status is wired into Vector's
```

File: website/content/en/highlights/2021-12-15-splunk-hec-improvements.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+forthcoming end-to-end acknowledgement feature. This feature will require that
+events be sent by sinks or persisted into disk buffers before sources will
+acknowledge them.
+
+### `splunk_hec` sink acknowledgements
+
+Previously, the `splunk_hec` sink simply treated successful HEC requests as the
+events being acknowledged by the Spunk receiver and so Vector would drop them
+from any buffers. Now, it is possible to configure the sink to wait until the
+Splunk reciever acknowledges the events via the [index
```
Comment (001wwang) [inline website/content/en/highlights/2021-12-15-splunk-hec-improvements.md:52]:
```suggestion
Splunk receiver acknowledges the events via the [index
```

File: website/content/en/highlights/2021-12-15-splunk-hec-improvements.md [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+```
+
+## Passthrough Token Routing
+
+Vector now also supports Splunk HEC token passthrough routing. When
+`store_hec_token` is enabled in a `splunk_hec` source, tokens included in
+requests to the source will be stored and used by downstream `splunk_hec` sinks.
+Any passed through token takes precedence over the `default_token` configuration
+set in the sink. With this update, you can now partition event batches in the
+`splunk_hec` sinks by token.
```
Comment (001wwang) [inline website/content/en/highlights/2021-12-15-splunk-hec-improvements.md:69]:
```suggestion
set in the sink.
```

**Review by 001wwang — CHANGES_REQUESTED** [review]
Nice! Suggested some changes to the wording and explanations that I think are worth making.

**Comment by jszwedko** [comment]
@001wwang thanks! I applied your suggestions.

**Review by 001wwang — APPROVED** [review]

**Pull request merged**